src/MLJXGBoostInterface.jl

```diff
@@ -1,770 +1,212 @@
-        warning *="Your `objective` suggests prediction of a "*
-        "`Finite` variable.\n You may want to consider XGBoostClassifier "*
-        "instead. "
     end
-    return warning
 end
 
-# For `XGBoost.DMatrix(Xmatrix, y)` `Xmatrix` must either be a julia `Array` or
-# a `SparseMatrixCSC` while `y` must be a `Vector` 
-_to_array(x::Union{Array, SparseArrays.SparseMatrixCSC}) = x
-_to_array(x::AbstractArray) = copyto!(similar(Array{eltype(x)}, axes(x)), x) 
-
-function MMI.fit(model::XGBoostRegressor
-             , verbosity::Int
-             , X
-             , y)
-
-             silent =
-                 verbosity > 0 ?  false : true
-    Xmatrix = _to_array(MMI.matrix(X))
-    dm = XGBoost.DMatrix(Xmatrix, label=_to_array(y))
-
-    objective =
-        model.objective in ["linear", "gamma", "tweedie"] ?
-            "reg:"*model.objective : model.objective
-
-    seed =
-        model.seed == -1 ? generate_seed() : model.seed
 
+eval(modelexpr(:XGBoostRegressor, :XGBoostAbstractRegressor, "reg:squarederror", :validate_reg_objective))
 
-    fitresult = XGBoost.xgboost(dm, model.num_round;
-                                kwargs(model, silent, seed, objective,model.eval_metric)...)
-
-    features = schema(X).names
-    importances = [named_importance(fi, features) for
-                   fi in XGBoost.importance(fitresult)]
-    cache = nothing
-
-    report = (feature_importances=importances, )
-
-    return fitresult, cache, report
-
-end
-
-
-function MMI.predict(model::XGBoostRegressor
-        , fitresult
-        , Xnew)
-    Xmatrix = _to_array(MMI.matrix(Xnew))
-    return XGBoost.predict(fitresult, Xmatrix)
+function kwargs(model, verbosity::Integer, obj)
+    excluded = [:importance_type]
+    fn = filter(∉(excluded), fieldnames(typeof(model)))
+    o = NamedTuple(n=>getfield(model, n) for n ∈ fn if !isnothing(getfield(model, n)))
+    o = merge(o, (silent=(verbosity ≤ 0),))
+    merge(o, (objective=_fix_objective(obj),))
 end
 
-
-## COUNT REGRESSOR
-
-mutable struct XGBoostCount <:MMI.Deterministic
-    num_round::Int
-    booster::String
-    disable_default_eval_metric::Int
-    eta::Float64
-    gamma::Float64
-    max_depth::Int
-    min_child_weight::Float64
-    max_delta_step::Float64
-    subsample::Float64
-    colsample_bytree::Float64
-    colsample_bylevel::Float64
-    lambda::Float64
-    alpha::Float64
-    tree_method::String
-    sketch_eps::Float64
-    scale_pos_weight::Float64
-    updater::String
-    refresh_leaf::Union{Int,Bool}
-    process_type::String
-    grow_policy::String
-    max_leaves::Int
-    max_bin::Int
-    predictor::String
-    sample_type::String
-    normalize_type::String
-    rate_drop::Float64
-    one_drop
-    skip_drop::Float64
-    feature_selector::String
-    top_k::Int
-    tweedie_variance_power::Float64
-    objective
-    base_score::Float64
-    eval_metric
-    seed::Int
-    nthread::Int
+function importances(X, r)
+    fs = schema(X).names
```

Comment:
```suggestion
    fs = Tables.schema(X).names
```

For clarity (to distinguish from `MLJModelnterface.schema`).

Reply:
This function wasn't doing anything anyway and has been removed.